discovery/amazon/lightsail.go

```diff
@@ -0,0 +1,235 @@
+	return NewLightsailDiscovery(c, opts.Logger), nil
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface for the Lightsail Config.
+func (c *LightsailSDConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	*c = LightsailDefaultSDConfig
+	type plain LightsailSDConfig
+	err := unmarshal((*plain)(c))
+	if err != nil {
+		return err
+	}
+	if c.Region == "" {
+		sess, err := session.NewSession()
+		if err != nil {
+			return err
+		}
+
+		metadata := ec2metadata.New(sess)
+
+		region, err := metadata.Region()
+		if err != nil {
+			return errors.New("Lightsail SD configuration requires a region")
+		}
+		c.Region = region
+	}
+	return nil
+}
+
+// Discovery periodically performs Lightsail-SD requests. It implements
+// the Discoverer interface.
+type LightsailDiscovery struct {
+	*refresh.Discovery
+	cfg       *LightsailSDConfig
+	lightsail *lightsail.Lightsail
+}
+
+// NewLightsailDiscovery returns a new LightsailDiscovery which periodically refreshes its targets.
+func NewLightsailDiscovery(conf *LightsailSDConfig, logger log.Logger) *LightsailDiscovery {
+	if logger == nil {
+		logger = log.NewNopLogger()
+	}
+	d := &LightsailDiscovery{
+		cfg: conf,
+	}
+	d.Discovery = refresh.NewDiscovery(
+		logger,
+		"lightsail",
+		time.Duration(d.cfg.RefreshInterval),
+		d.refresh,
+	)
+	return d
+}
+
+func (d *LightsailDiscovery) lightsailClient() (*lightsail.Lightsail, error) {
+	if d.lightsail != nil {
+		return d.lightsail, nil
+	}
+
+	creds := credentials.NewStaticCredentials(d.cfg.AccessKey, string(d.cfg.SecretKey), "")
+	if d.cfg.AccessKey == "" && d.cfg.SecretKey == "" {
+		creds = nil
+	}
+
+	sess, err := session.NewSessionWithOptions(session.Options{
+		Config: aws.Config{
+			Endpoint:    &d.cfg.Endpoint,
+			Region:      &d.cfg.Region,
+			Credentials: creds,
+		},
+		Profile: d.cfg.Profile,
+	})
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create aws session")
+	}
+
+	if d.cfg.RoleARN != "" {
+		creds := stscreds.NewCredentials(sess, d.cfg.RoleARN)
+		d.lightsail = lightsail.New(sess, &aws.Config{Credentials: creds})
+	} else {
+		d.lightsail = lightsail.New(sess)
+	}
+
+	return d.lightsail, nil
+}
+
+func (d *LightsailDiscovery) refresh(ctx context.Context) ([]*targetgroup.Group, error) {
+	lightsailClient, err := d.lightsailClient()
+	if err != nil {
+		return nil, err
+	}
+
+	tg := &targetgroup.Group{
+		Source: d.cfg.Region,
+	}
+
+	input := &lightsail.GetInstancesInput{}
+
+	output, err := lightsailClient.GetInstancesWithContext(ctx, input)
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok && (awsErr.Code() == "AuthFailure" || awsErr.Code() == "UnauthorizedOperation") {
+			d.lightsail = nil
+		}
+		return nil, errors.Wrap(err, "could not get instances")
+	}
+
+	for _, inst := range output.Instances {
+		if inst.PrivateIpAddress == nil {
+			continue
+		}
+
+		labels := model.LabelSet{
+			lightsailLabelInstanceSupportCode: model.LabelValue(*inst.SupportCode),
```

Comment:
Can we set the other here? like IP,  blueprint etc..

Reply:
https://github.com/prometheus/prometheus/pull/8693/commits/45b8e9f303661f48eda638d9ad9fee65ee6aec95
Do we want any more labels inside here?